File: src/main/java/wraith/fabricaeexnihilo/compatibility/rei/crucible/CrucibleHeatDisplay.java
package wraith.fabricaeexnihilo.compatibility.rei.crucible;

import me.shedaniel.rei.api.common.category.CategoryIdentifier;
import me.shedaniel.rei.api.common.display.Display;
import me.shedaniel.rei.api.common.entry.EntryIngredient;
import net.minecraft.util.Identifier;
import wraith.fabricaeexnihilo.compatibility.rei.PluginEntry;
import wraith.fabricaeexnihilo.recipe.crucible.CrucibleHeatRecipe;

import java.util.List;
import java.util.Optional;

public class CrucibleHeatDisplay implements Display {
    public final int heat;
    public final EntryIngredient source;
    private final Identifier id;

    public CrucibleHeatDisplay(CrucibleHeatRecipe recipe) {
        this.source = recipe.getBlock().asReiIngredient();
        this.heat = recipe.getHeat();
        this.id = recipe.getId();
    }

    @Override
    public CategoryIdentifier<?> getCategoryIdentifier() {
        return PluginEntry.HEATING;
    }

    @Override
    public List<EntryIngredient> getInputEntries() {
        return List.of(source);
    }

    @Override
    public List<EntryIngredient> getOutputEntries() {
        return List.of();
    }

    @Override
    public Optional<Identifier> getDisplayLocation() {
        return Optional.of(id);
    }
}

File: src/main/kotlin/exnihilofabrico/compatibility/rei/crucible/CrucibleHeatDisplay.kt
package exnihilofabrico.compatibility.rei.crucible

import exnihilofabrico.api.recipes.crucible.CrucibleHeatRecipe
import exnihilofabrico.compatibility.rei.PluginEntry
import exnihilofabrico.util.asREIEntry
import exnihilofabrico.util.getExNihiloItemStack
import me.shedaniel.rei.api.EntryStack
import me.shedaniel.rei.api.RecipeDisplay

class CrucibleHeatDisplay(val recipe: CrucibleHeatRecipe): RecipeDisplay {
    override fun getRecipeCategory() = PluginEntry.CRUCIBLE_HEAT

    override fun getOutputEntries() = listOf<EntryStack>()
    override fun getInputEntries() =
        listOf(
            listOf(recipe.ingredient.asREIEntries(), recipe.fluid.asREIEntries()).flatten(),
            listOf(getExNihiloItemStack("stone_crucible").asREIEntry()))
    fun getHeat() = recipe.value

}